pallets/ethereum-transaction/src/lib.rs

```diff
@@ -32,106 +29,45 @@ mod tests;
 
 #[frame_support::pallet]
 pub mod pallet {
+	use frame_system::pallet_prelude::OriginFor;
+
 	use super::*;
 
 	#[pallet::pallet]
 	#[pallet::generate_store(pub (super) trait Store)]
 	pub struct Pallet<T>(_);
 
 	#[pallet::config]
-	pub trait Config: frame_system::Config + pallet_ethereum::Config {
-		/// The event type.
-		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+	pub trait Config: frame_system::Config + pallet_ethereum::Config
+	where
+		OriginFor<Self>:
+			From<pallet_ethereum::Origin> + Into<Result<pallet_ethereum::Origin, OriginFor<Self>>>,
+	{
 	}
 
 	/// Storage for nonce.
 	#[pallet::storage]
 	pub(crate) type Nonce<T: Config> = StorageValue<_, U256, ValueQuery>;
 
-	#[pallet::event]
-	#[pallet::generate_deposit(pub (super) fn deposit_event)]
-	pub enum Event<T: Config> {
-		/// A call was executed.
-		Executed {
-			from: H160,
-			to: H160,
-			exit_reason: ExitReason,
-			value: Vec<u8>,
-		},
-	}
-
-	#[pallet::error]
-	pub enum Error<T> {
-		/// Trying to pop from an empty stack.
-		StackUnderflow,
-
-		/// Trying to push into a stack over stack limit.
-		StackOverflow,
-
-		/// Jump destination is invalid.
-		InvalidJump,
-
-		/// An opcode accesses memory region, but the region is invalid.
-		InvalidRange,
-
-		/// Encountered the designated invalid opcode.
-		DesignatedInvalid,
-
-		/// Call stack is too deep (runtime).
-		CallTooDeep,
-
-		/// Create opcode encountered collision (runtime).
-		CreateCollision,
-
-		/// Create init code exceeds limit (runtime).
-		CreateContractLimit,
-
-		/// Invalid opcode during execution or starting byte is 0xef. See [EIP-3541](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-3541.md).
-		InvalidCode(u8),
-
-		/// An opcode accesses external information, but the request is off
-		/// offset limit (runtime).
-		OutOfOffset,
-
-		/// Execution runs out of gas (runtime).
-		OutOfGas,
-
-		/// Not enough fund to start the execution (runtime).
-		OutOfFund,
-
-		/// PC underflowed (unused).
-		PCUnderflow,
-
-		/// Attempt to create an empty account (runtime, unused).
-		CreateEmpty,
-
-		/// The operation is not supported.
-		NotSupported,
-		/// The trap (interrupt) is unhandled.
-		UnhandledInterrupt,
-
-		/// Machine encountered an explicit revert.
-		Reverted,
-
-		/// Unexpected result when executing a transaction.
-		UnexpectedExecuteResult,
-
-		/// Other normal errors.
-		Other,
-	}
-
-	impl<T: Config> Pallet<T> {
+	impl<T: Config> Pallet<T>
+	where
+		OriginFor<T>:
+			From<pallet_ethereum::Origin> + Into<Result<pallet_ethereum::Origin, OriginFor<T>>>,
+	{
 		fn get_transaction_signature() -> Option<TransactionSignature> {
-			//TODO(cdamian): Same signature as the one in ethereum-xcm.
 			TransactionSignature::new(
 				TRANSACTION_RECOVERY_ID,
-				H256::from_low_u64_be(1u64),
-				H256::from_low_u64_be(1u64),
+				H256::from_low_u64_be(2u64),
```

Comment:
Just want to make sure this bump is intended? If you have the time, would be interested what this number reflects as the type is quite opaque:

```rust
pub struct TransactionSignature {
	v: TransactionRecoveryId,
	r: H256,
	s: H256,
}
```

Reply:
I just wanted the signature to not match the one from `pallet-xcm-transactor`. I have to be honest, no clue whether this makes snese. The signature is not checked, so I guess it is not really relevant.

Reply:
All right, thanks nevertheless!